src/services/package-manager/batch-import-export-service.ts

```diff
@@ -0,0 +1,59 @@
+import {batchImportExportApi} from "../../api/batch-import-export-api";
+import {logger} from "../../util/logger";
+import {v4 as uuidv4} from "uuid";
+import {PackageExportTransport} from "../../interfaces/package-export-transport";
+import {FileService, fileService} from "../file-service";
+import {packageApi} from "../../api/package-api";
+import {variableService} from "./variable-service";
+import {PackageManagerVariableType} from "../../interfaces/package-manager.interfaces";
+import {dataModelService} from "./datamodel-service";
+
+class BatchImportExportService {
+
+    public async listActivePackages(flavors: string[]): Promise<void> {
+        const activePackages = await batchImportExportApi.findAllActivePackages(flavors);
+        activePackages.forEach(pkg => {
+            logger.info(`${pkg.name} - Key: "${pkg.key}"`)
+        });
+    }
+
+    public async findAndExportListOfActivePackages(flavors: string[], packageKeys: string[], withDependencies: boolean): Promise<void> {
+        let packagesToExport: PackageExportTransport[];
+
+        if (packageKeys.length) {
+            packagesToExport = await batchImportExportApi.findActivePackagesByKeys(packageKeys, withDependencies);
+        } else  {
+            packagesToExport = await batchImportExportApi.findAllActivePackages(flavors, withDependencies);
+        }
+
+        const packageByKey = new Map<string, PackageExportTransport>();
+        packagesToExport.forEach(pkg => packageByKey.set(pkg.key, pkg));
+
+        if (withDependencies) {
+            const studioPackageWithDataModelVariableAssignments = await variableService.getVariableAssignmentsForNodes(PackageManagerVariableType.DATA_MODEL);
+            const dataModelDetailsByNode = await dataModelService.getDataModelDetailsForPackages(studioPackageWithDataModelVariableAssignments);
+            studioPackageWithDataModelVariableAssignments.forEach(studioPackage => {
+                if (packageByKey.has(studioPackage.key)) {
+                    packageByKey.get(studioPackage.key).datamodels = dataModelDetailsByNode.get(studioPackage.key);
+                }
+            });
+        }
+
+        const studioPackages = await packageApi.findAllPackages();
```

Comment:
can we check if we can call by keys here, so we don't fetch the whole studio packages?

Reply:
There is no API for getting a list of packages by keys. We can refactor the code to reuse packages with variables that we get [here](https://github.com/celonis/content-cli/pull/165/files#diff-10b781e4b0f7b649247dbfc2845e42dd87ef2b6ea9f130b055c3675655151d2dR25) or add a call to [get a single package with variables](https://www.celonis.dev/catalog/default/api/cloud-package-manager-package-manager-api/definition#/package/findPackageWithVariableAssignments). What do you think?

Reply:
We can reuse those then, by fetching them once and passing on the list in the other two dependent packages.